java/gradle.htmlui/src/org/netbeans/modules/gradle/htmlui/GradleArchetype.java

```diff
@@ -18,59 +18,74 @@
  */
 package org.netbeans.modules.gradle.htmlui;
 
-import java.io.IOException;
+import java.io.File;
 import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
 import java.util.Map;
-import org.netbeans.api.project.ProjectManager;
-import org.netbeans.api.templates.FileBuilder;
-import org.netbeans.api.templates.FileBuilder.Mode;
+import org.netbeans.modules.gradle.spi.newproject.TemplateOperation;
 import org.openide.filesystems.FileObject;
 import org.openide.filesystems.FileUtil;
 import org.openide.util.MapFormat;
 
 public final class GradleArchetype {
     private final FileObject templates;
-    private final FileObject projectFo;
```

Comment:
I always prefer the `FileObject` abstraction over using plain `File`. It makes me sad when I see code moving to the opposite direction.

What is the reason for making such change? The fact that modifications using `File` aren't notified as soon as they are made to various listeners?

Reply:
The reason using File object during the template steps, as they are non-existent ones while the operation steps are collected, so in that case FileObjects would be null.
The templating has two major phase: collecting the steps to be executed and then execute the steps.